README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# mali: Datasets for Use in Designing Surveys in Mali <img src="man/figures/mali.png" width="200" align="right" />

[![Project Status: Active – The project has reached a stable, usable
state and is being actively
developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
[![lifecycle](https://img.shields.io/badge/lifecycle-maturing-blue.svg)](https://www.tidyverse.org/lifecycle/#maturing)
[![Travis build
status](https://travis-ci.org/validmeasures/mali.svg?branch=master)](https://travis-ci.org/validmeasures/mali)
[![AppVeyor build
status](https://ci.appveyor.com/api/projects/status/github/validmeasures/mali?branch=master&svg=true)](https://ci.appveyor.com/project/validmeasures/mali)

Designing surveys require relevant datasets to be used as basis for
sample size calculations, sampling design, survey planning/logistics and
survey implementation. These include datasets on population, lists of
sampling clusters, map datasets for spatial sampling, and previous
survey datasets that can be used for estimating indicator variance and
design effects. This package contains relevant datasets for use in
designing surveys in Mali

## Installation

`mali` is not yet released on [CRAN](https://cran.r-project.org).

You can install the development version of `mali` from
[GitHub](https://github.com/validmeasures/mali) with:

``` r
if(!require(devtools)) install.packages("devtools") 
install_github("validmeasures/mali")
```

## Usage

### Accessing Mali map data

The following packages will be required when accessing and manipulating
map data available from the `mali` package.

``` r
require(rgdal)
require(rgeos)
require(raster)
```

#### Country borders

To access country borders, use the following command:

``` r
mali::mli0
#> class       : SpatialPolygonsDataFrame 
#> features    : 1 
#> extent      : -12.23924, 4.24467, 10.14137, 24.99951  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 11
#> names       : OBJECTID_1, admin0Name, admin0Pcod, admin0RefN, admin0AltN, admin0Al_1,       date,    validOn, validTo, Shape_Leng, Shape_Area 
#> value       :          1,       Mali,         ML,       Mali,         NA,         NA, 2012/12/31, 2015/04/08,      NA,   71.76874,   106.8238
```

The output provides overview information on the features included in the
country borders map and the variables included in the country map data.
This is a summary output for a `SpatialPolygonsDataFrame`.

To plot the country borders, the `plot()` function from the `sp` package
can be used as follows:

``` r
plot(mali::mli0)
```

<img src="man/figures/README-country-plot-1.png" width="100%" />

#### Region borders

To access region borders, use the following command:

``` r
mali::mli1
#> class       : SpatialPolygonsDataFrame 
#> features    : 9 
#> extent      : -12.23924, 4.24467, 10.14137, 24.99951  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 13
#> names       : OBJECTID, admin0Name, admin0Pcod, admin1Name, admin1Pcod, admin1RefN, admin1AltN, admin1Al_1,       date,    validOn, validTo, Shape_Leng,  Shape_Area 
#> min values  :        1,       Mali,         ML,     Bamako,       ML01,     Bamako,         NA,         NA, 2015/04/08, 2015/04/08,      NA,  0.7517338,  0.02043903 
#> max values  :        9,       Mali,         ML, Tombouctou,       ML09, Tombouctou,         NA,         NA, 2015/04/08, 2015/04/08,      NA, 33.7798722, 43.15657700
```

The output provides overview information on the features included in the
region borders map and the variables included in the region map data.
This is a summary output for a `SpatialPolygonsDataFrame`.

To plot the region borders, the `plot()` function from the `sp` package
can be used as follows:

``` r
plot(mali::mli1)
```

<img src="man/figures/README-region-plot-1.png" width="100%" />

#### Cercle borders

To access cercle borders, use the following command:

``` r
mali::mli2
#> class       : SpatialPolygonsDataFrame 
#> features    : 50 
#> extent      : -12.23924, 4.24467, 10.14137, 24.99951  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 15
#> names       : OBJECTID, admin0Name, admin0Pcod, admin1Name, admin1Pcod, admin2Name, admin2Pcod, admin2RefN, admin2AltN, admin2Al_1,       date,    validOn, validTo, Shape_Leng,  Shape_Area 
#> min values  :        1,       Mali,         ML,     Bamako,       ML01,   Abeibara,     ML0101,   Abeibara,         NA,         NA, 2015/04/08, 2015/04/08,      NA,  0.7517338,  0.02043903 
#> max values  :        9,       Mali,         ML, Tombouctou,       ML09,   Youwarou,     ML0901,   Youwarou,         NA,         NA, 2015/04/08, 2015/04/08,      NA, 26.0765915, 28.80383259
```

The output provides overview information on the features included in the
cercle borders map and the variables included in the cercle map data.
This is a summary output for a `SpatialPolygonsDataFrame`.

To plot the cercle borders, the `plot()` function from the `sp` package
can be used as follows:

``` r
plot(mali::mli2)
```

<img src="man/figures/README-cercle-plot-1.png" width="100%" />

#### Commune borders

To access commune borders, use the following command:

``` r
mali::mli3
#> class       : SpatialPolygonsDataFrame 
#> features    : 701 
#> extent      : -12.23924, 4.24467, 10.14137, 24.99951  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 17
#> names       : OBJECTID_1, admin0Name, admin0Pcod, admin1Name, admin1Pcod, admin2Name, admin2Pcod, admin3Name, admin3Pcod, admin3RefN, admin3AltN, admin3Al_1,       date,    validOn, validTo, ... 
#> min values  :          1,       Mali,         ML,     Bamako,       ML01,   Abeibara,     ML0101,   Abeibara,   ML010101,   Abeibara,         NA,         NA, 2013/05/31, 2015/04/08,      NA, ... 
#> max values  :         99,       Mali,         ML, Tombouctou,       ML09,   Youwarou,     ML0901,     Zegoua,   ML090106,     Zegoua,         NA,         NA, 2013/05/31, 2015/04/08,      NA, ...
```

The output provides overview information on the features included in the
commune borders map and the variables included in the commune map data.
This is a summary output for a `SpatialPolygonsDataFrame`.

To plot the commune borders, the `plot()` function from the `sp` package
can be used as follows:

``` r
plot(mali::mli3)
```

<img src="man/figures/README-commune-plot-1.png" width="100%" />

#### Bamako borders

To access Bamako borders, use the following command:

``` r
mali::bamako
#> class       : SpatialPolygonsDataFrame 
#> features    : 71 
#> extent      : -8.081477, -7.91215, 12.50289, 12.71143  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
#> variables   : 19
#> names       : OBJECTID, admin0Name, admin0Pcod, admin1Name, admin1Pcod, admin2Name, admin2Pcod, admin3Name, admin3Pcod,        admin4Name,  admin4Pcod,        admin4RefN,      admin4AltN, admin4Al_1,       date, ... 
#> min values  :        1,       Mali,         ML,     Bamako,       ML09,     Bamako,     ML0901,  Commune I,   ML090101,   Babouillabougou, ML090101001,   Babouillabougou, Babouillabougou,         NA, 2013/05/31, ... 
#> max values  :        9,       Mali,         ML,     Bamako,       ML09,     Bamako,     ML0901, Commune VI,   ML090106, Zone Industrielle, ML090106903, Zone Industrielle, Zone industriel,         NA, 2013/05/31, ...
```

The output provides overview information on the features included in the
bamako borders map and the variables included in the bamako map data.
This is a summary output for a `SpatialPolygonsDataFrame`.

To plot the bamako borders, the `plot()` function from the `sp` package
can be used as follows:

``` r
plot(mali::bamako)
```

<img src="man/figures/README-bamako-plot-1.png" width="100%" />


NEWS.md
# mali 0.1.0

* Added a `NEWS.md` file to track changes to the package.


data-raw/processMaps.R
################################################################################
#
# Setup
#
################################################################################

## Load libraries
if(!require(rgdal)) install.packages("rgdal")
if(!require(rgeos)) install.packages("rgeos")

if(!require(devtools)) install.packages("devtools")
if(!require(spatialsampler)) install_github("SpatialWorks/spatialsampler")
if(!require(gadmr)) install_github("SpatialWorks/gadmr")

if(!require(stringr)) install.packages("stringr")

options(stringsAsFactors = FALSE)

################################################################################
#
# Read Mali maps from OCHA/Humanitarian data
#
################################################################################

## Country borders
mli0 <- readOGR(dsn = "data-raw/maps/mli_admbnda_adm0_gov",
                layer = "mli_admbnda_adm0_gov",
                verbose = FALSE)

## Save country borders as GPKG
writeOGR(obj = mli0,
         dsn = "data-raw/maps/mali.gpkg",
         layer = "country",
         driver = "GPKG")

## Save country borders data in data package
use_data(mli0, overwrite = TRUE)

## Region borders
mli1 <- readOGR(dsn = "data-raw/maps/mli_admbnda_adm1_gov_ocha_itos",
                layer = "mli_admbnda_adm1_gov_ocha_itos",
                verbose = FALSE)

## Save country borders as GPKG
writeOGR(obj = mli1,
         dsn = "data-raw/maps/mali.gpkg",
         layer = "region",
         driver = "GPKG")

## Save country borders data in data package
use_data(mli1, overwrite = TRUE)

## Cercles borders
mli2 <- readOGR(dsn = "data-raw/maps/mli_admbnda_adm2_gov_ocha_itos",
                layer = "mli_admbnda_adm2_gov_ocha_itos",
                verbose = FALSE)

## Save cercles borders as GPKG
writeOGR(obj = mli2,
         dsn = "data-raw/maps/mali.gpkg",
         layer = "cercle",
         driver = "GPKG")

## Save cercles borders data in data package
use_data(mli2, overwrite = TRUE)

## Commune borders
mli3 <- readOGR(dsn = "data-raw/maps/mli_admbnda_adm3_gov_ocha",
                layer = "mli_admbnda_adm3_gov_ocha",
                verbose = FALSE)

## Save commune borders as GPKG
writeOGR(obj = mli3,
         dsn = "data-raw/maps/mali.gpkg",
         layer = "commune",
         driver = "GPKG")

## Save commune borders data as data in package
use_data(mli3, overwrite = TRUE)

## Bamako borders
bamako <- readOGR(dsn = "data-raw/maps/mli_admbnda_adm4_iom_gov",
                  layer = "mli_admbnda_adm4_iom_gov",
                  verbose = FALSE)

## Save Bamako borders as GPKG
writeOGR(obj = bamako,
         dsn = "data-raw/maps/bamako.gpkg",
         layer = "bamako",
         driver = "GPKG")

x <- str_replace(bamako@data$admin4Name, pattern = "é", "e")
x <- str_replace(x, pattern = "é", "e")
x <- str_replace(x, pattern = "è", "e")
x <- str_replace(x, pattern = "è", "e")

bamako@data$admin4Name <- x

## Save Bamako borders data as data in package
use_data(bamako, overwrite = TRUE)

## Settlements
settlements <- readOGR(dsn = "data-raw/maps/MLI_PopulatedPlaces_SHP",
                       layer = "COD_MLI_PopulatedPlaces_Rena",
                       verbose = FALSE)

writeOGR(obj = settlements,
         dsn = "data-raw/maps/mali.gpkg",
         layer = "settlements",
         driver = "GPKG")

settlementsCoords <- data.frame(settlements@coords)
names(settlementsCoords) <- c("longitude", "latitude")

settlements <- data.frame(settlements@data, settlementsCoords)
write.csv(settlements, "data-raw/maps/settlements.csv", row.names = FALSE)

settlements$featureNam <- str_replace(settlements$featureNam, pattern = "é", replacement = "e")

use_data(settlements, overwrite = TRUE)


R/data.R
################################################################################
#
#' mali
#'
#' Datasets for Use in Designing Surveys in Mali
#'
#' Designing surveys require relevant datasets to be used as basis for
#' sample size calculations, sampling design, survey planning/logistics and
#' survey implementation. These include datasets on population, lists of sampling
#' clusters, map datasets for spatial sampling, and previous survey datasets that
#' can be used for estimating indicator variance and design effects. This package
#' contains relevant datasets for use in designing surveys in Mali.
#'
#' @docType package
#' @name mali
#'
#
################################################################################
NULL

################################################################################
#
#' mli0
#'
#' Administrative Level 0 (Country) map of Mali
#'
#' @format A SpatialPolygonsDataFrame with 1 feature:
#' \describe{
#' \item{\code{OBJECTID}}{Object identifier}
#' \item{\code{admin0Name}}{Name of administrative level 0 (country)}
#' \item{\code{admin0Pcod}}{Postal code of administrative level 0 (country)}
#' \item{\code{admin0RefN}}{Reference name of administrative level 0 (country)}
#' \item{\code{admin0AltN}}{Alternative name of administrative level 0 (country)}
#' \item{\code{admin0Al_1}}{Alternative name of administrative level 0 (country)}
#' \item{\code{date}}{Date map was poduced}
#' \item{\code{validOn}}{Date map was last validated}
#' \item{\code{validTo}}{Date map is valid up to}
#' \item{\code{Shape_Leng}}{Lenght of shape}
#' \item{\code{Shape_Area}}{Shape of area}
#' }
#'
#' @source Humanitarian Data Exchange <https://data.humdata.org/dataset/administrative-boundaries-cod-mli>
#
################################################################################
"mli0"


################################################################################
#
#' mli1
#'
#' Administrative Level 1 (Region) map of Mali
#'
#' @format A SpatialPolygonsDataFrame with 9 features:
#' \describe{
#' \item{\code{OBJECTID}}{Object identifier}
#' \item{\code{admin1Name}}{Name of administrative level 1 (region)}
#' \item{\code{admin1Pcod}}{Postal code of administrative level 1 (region)}
#' \item{\code{admin1RefN}}{Reference name of administrative level 1 (region)}
#' \item{\code{admin1AltN}}{Alternative name of administrative level 1 (region)}
#' \item{\code{admin1Al_1}}{Alternative name of administrative level 1 (region)}
#' \item{\code{admin0Name}}{Country (administrative level 0) name}
#' \item{\code{admin0Pcod}}{Country (administrative level 1) code}
#' \item{\code{date}}{Date map was poduced}
#' \item{\code{validOn}}{Date map was last validated}
#' \item{\code{validTo}}{Date map is valid up to}
#' \item{\code{Shape_Leng}}{Lenght of shape}
#' \item{\code{Shape_Area}}{Shape of area}
#' }
#'
#' @source Humanitarian Data Exchange <https://data.humdata.org/dataset/administrative-boundaries-cod-mli>
#
################################################################################
"mli1"


################################################################################
#
#' mli2
#'
#' Administrative Level 2 (Cercle) map of Mali
#'
#' @format A SpatialPolygonsDataFrame with 50 features:
#' \describe{
#' \item{\code{OBJECTID}}{Object identifier}
#' \item{\code{admin2Name}}{Name of administrative level 2 (cercle)}
#' \item{\code{admin2Pcode}}{Postal code of administrative level 2 (cercle)}
#' \item{\code{admin2RefN}}{Reference name of administrative level 2 (cercle)}
#' \item{\code{admin2AltN}}{Alternative name of administrative level 2 (cercle)}
#' \item{\code{admin2Al_1}}{Alternative name of administrative level 2 (cercle)}
#' \item{\code{admin1Name}}{Name of administrative level 1 (region)}
#' \item{\code{admin1Pcod}}{Postal code of administrative level 1 (region)}
#' \item{\code{admin0Name}}{Country (administrative level 0) name}
#' \item{\code{admin0Pcod}}{Country (administrative level 1) code}
#' \item{\code{date}}{Date map was poduced}
#' \item{\code{validOn}}{Date map was last validated}
#' \item{\code{validTo}}{Date map is valid up to}
#' \item{\code{Shape_Leng}}{Length of shape}
#' \item{\code{Shape_Area}}{Area of shape}
#' }
#'
#' @source Humanitarian Data Exchange <https://data.humdata.org/dataset/administrative-boundaries-cod-mli>
#
################################################################################
"mli2"


################################################################################
#
#' mli3
#'
#' Administrative Level 3 (Commune) map of Mali
#'
#' @format A SpatialPolygonsDataFrame with 701 features:
#' \describe{
#' \item{\code{OBJECTID_1}}{Object identifier}
#' \item{\code{admin0Name}}{Country (administrative level 0) name}
#' \item{\code{admin0Pcod}}{Country (administrative level 1) code}
#' \item{\code{admin1Name}}{Name of administrative level 1 (region)}
#' \item{\code{admin1Pcod}}{Postal code of administrative level 1 (region)}
#' \item{\code{admin2Name}}{Name of administrative level 2 (cercle)}
#' \item{\code{admin2Pcod}}{Postal code of administrative level 2 (cercle)}
#' \item{\code{admin3Name}}{Name of administrative level 3 (commune)}
#' \item{\code{admin3Pcod}}{Postal code of administrative level 3 (commune)}
#' \item{\code{admin3RefN}}{Reference name of administrative level 3 (commune)}
#' \item{\code{admin3AltN}}{Alternative name of administrative level 3 (commune)}
#' \item{\code{admin3Al_1}}{Alternative name of administrative level 3 (commune)}
#' \item{\code{date}}{Date map was poduced}
#' \item{\code{validOn}}{Date map was last validated}
#' \item{\code{validTo}}{Date map is valid up to}
#' \item{\code{Shape_Leng}}{Length of shape}
#' \item{\code{Shape_Area}}{Area of shape}
#' }
#'
#' @source Humanitarian Data Exchange <https://data.humdata.org/dataset/administrative-boundaries-cod-mli>
#
################################################################################
"mli3"


################################################################################
#
#' bamako
#'
#' Bamako District Capital Map
#'
#' @format A SpatialPolygonsDataFrame with 71 features:
#' \describe{
#' \item{\code{OBJECTID_1}}{Object identifier}
#' \item{\code{admin0Name}}{Country (administrative level 0) name}
#' \item{\code{admin0Pcod}}{Country (administrative level 1) code}
#' \item{\code{admin1Name}}{Name of administrative level 1 (region)}
#' \item{\code{admin1Pcod}}{Postal code of administrative level 1 (region)}
#' \item{\code{admin2Name}}{Name of administrative level 2 (cercle)}
#' \item{\code{admin2Pcod}}{Postal code of administrative level 2 (cercle)}
#' \item{\code{admin3Name}}{Name of administrative level 3 (commune)}
#' \item{\code{admin3Pcod}}{Postal code of administrative level 3 (commune)}
#' \item{\code{admin4Name}}{Name of Bamako city block}
#' \item{\code{admin4Pcod}}{Postal code of Bamako city block}
#' \item{\code{admin4RefN}}{Reference name of Bamako city block}
#' \item{\code{admin4AltN}}{Alternative name of Bamako city block}
#' \item{\code{admin4Al_1}}{Alternative name of Bamako city block}
#' \item{\code{date}}{Date map was poduced}
#' \item{\code{validOn}}{Date map was last validated}
#' \item{\code{validTo}}{Date map is valid up to}
#' \item{\code{Shape_Leng}}{Length of shape}
#' \item{\code{Shape_Area}}{Area of shape}
#' }
#'
#' @source Humanitarian Data Exchange <https://data.humdata.org/dataset/administrative-boundaries-cod-mli>
#
################################################################################
"bamako"


################################################################################
#
#' settlements
#'
#' Settlements/villages of Mali
#'
#' @format A data.frame with 20 columns and 18769 rows:
#' \describe{
#' \item{\code{featureNam}}{Name of settlement/village}
#' \item{\code{pcode}}{Settlement/village postal code}
#' \item{\code{featureRef}}{Reference name of settlement/village}
#' \item{\code{featureAlt}}{Alternative name of settlement/village}
#' \item{\code{popPlaceCl}}{Classification of settlement}
#' \item{\code{admin0Name}}{Country (administrative level 0) name}
#' \item{\code{admin0Pcod}}{Country (administrative level 1) code}
#' \item{\code{admin1Name}}{Name of administrative level 1 (region)}
#' \item{\code{admin1Pcod}}{Postal code of administrative level 1 (region)}
#' \item{\code{admin2Name}}{Name of administrative level 2 (cercle)}
#' \item{\code{admin2Pcod}}{Postal code of administrative level 2 (cercle)}
#' \item{\code{admin3Name}}{Name of administrative level 3 (commune)}
#' \item{\code{admin3Pcod}}{Postal code of administrative level 3 (commune)}
#' \item{\code{date}}{Date map was poduced}
#' \item{\code{validOn}}{Date map was last validated}
#' \item{\code{validTo}}{Date map is valid up to}
#' \item{\code{longitude}}{Longitude coordinate of settlement/village}
#' \item{\code{Latitude}}{Latitude coordinate of settlement/village}
#' }
#'
#' @source Humanitarian Data Exchange <https://data.humdata.org/dataset/administrative-boundaries-cod-mli>
#
################################################################################
"settlements"
